### DISCUSSION TITLE
[RESOLVED] Error when running puzzletokens not directly from sketch.

### DISCUSSION THREAD

**Discussion by bmooij-lg** [ingrammicro/puzzle-tokens#52] [Q&A] [answered]
Since recently I'm getting the following error when my script tries to open, style and save a sketch file using puzzletokens:
"TypeError: undefined is not an object (evaluating 'this.pathToSource.substring')"

This has worked fine for quite a while, and I haven't changed the code or setup since to my knowledge.

I've got shell script to run it as follows:
/Applications/Sketch.app/Contents/Resources/sketchtool/bin/sketchtool --without-activating=YES --new-instance=No run ~/Library/Application\ Support/com.bohemiancoding.sketch3/Plugins/PuzzleTokens.sketchplugin "cmdRun" --context=$context
with
context="${context},\"commands\":\"apply,save,close\"}"

running the same .less from sketch directly works fine though...
Let me know if this isn't a plugin issue, but I'm having trouble understanding what's causing this.

**Comment by MaxBazarov** [comment]
Investigating….

**Comment by MaxBazarov** [comment, marked as answer]
@bmooij-lg  Fixed in 8.19.1
Please confirm

**Reply by bmooij-lg** [reply]
Yes, it's working again, thank you!

**Reply by MaxBazarov** [reply]
Sorry for the broken functionality.
I added a new feature and didn't test all functions  — my fault.

BTW, now can use new command - "Apply to existing styles". You select a file with token values which used before to apply styles. The plugin gets tokens and update all related styles. It does it very quickly (1 min vs 15 min on my main project).  Works very will to switch between skins.

It requires this setting enabled - 
<img width="594" alt="Screenshot 2022-05-18 at 00 29 13" src="https://user-images.githubusercontent.com/18418306/168913276-070fd374-c2e5-4202-bb15-14de1ac07383.png">
